src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -24,8 +24,11 @@ public partial class Socket : IDisposable
 
         private SafeSocketHandle _handle;
 
-        // _rightEndPoint is null if the socket has not been bound.  Otherwise, it is any EndPoint of the
-        // correct type (IPEndPoint, etc).
+        // _rightEndPoint is null if the socket has not been bound.  Otherwise, it is an EndPoint of the
+        // correct type (IPEndPoint, etc). The Bind operation sets _rightEndPoint. Other operations must only set
+        // it when the value is still null.
+        // This enables tracking the file created by UnixDomainSocketEndPoint when the Socket is bound,
+        // and to delete that file when the Socket gets disposed.
```

Comment:
I avoided adding a field to `Socket` for the stake of tracking the file to be deleted. Instead this is tracked through `_rightEndPoint` `UnixDomainSocketEndPoint.BoundFileName`.
On Bind an instance is created that has this property set. And on dispose, the file then gets deleted.

Reply:
I've wondered this for a while... what does the "right" part of the name mean?

Reply:
Just guessing: _right_ type. The name is not very clear. Maybe we should change it to `_typedEndpoint` or `_endpointType`?

Reply:
I keep asking this question every time I see that variable.

Reply:
I don't really know either. Would love to understand.

It might be "right" in the sense that it's the result after resolving wildcards like port=0. Does that seem right?